### PR TITLE
feat: docs for kubernetes resource check & kubernetes check

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/docs/issues/198

### CODE REVIEW

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  ]}
+/>
+
+### Templating
+
+The static and non static resources can be templated. This is helpful to create resources with random names. Example:
+you can setup a resource to create a pod with random name on each check run. This way you don't have to wait for the pod to be deleted on every check.
+
+:::info
+Templating the **Group**, **Version**, **Kind** & **Namespace** however isn't allowed.
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:130]:
```suggestion
Templating the **Group**, **Version**, **Kind** & **Namespace** fields is not supported on `staticResources` 
```

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        'Interval to check if all static & non-static resources are ready. _(Default: 5s)_',
+      scheme: 'Duration'
+    }
+  ]}
+/>
+
+### Templating
+
+The static and non static resources can be templated. This is helpful to create resources with random names. Example:
+you can setup a resource to create a pod with random name on each check run. This way you don't have to wait for the pod to be deleted on every check.
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:127]: Can we create this example using a details section

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  connection="kubeconfig"
+  rows={[
+    {
+      field: 'resources',
+      description: 'Manifests that should be applied',
+      scheme: '`[]KubernetesManifest`',
+      required: true
+    },
+    {
+      field: 'static resources',
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:32]:
```suggestion
      field: 'staticResources',
```

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      description:
+        'Interval to check if all static & non-static resources are ready. _(Default: 5s)_',
+      scheme: 'Duration'
+    }
+  ]}
+/>
+
+### Templating
+
+The static and non static resources can be templated. This is helpful to create resources with random names. Example:
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:126]:
```suggestion
The `resources` and `staticResources` fields can be templated using Go Text Templates. This is helpful to create resources with random names. Example:
```

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+### Wait For
+
+<Fields
+  rows={[
+    {
+      field: 'expr',
+      description: `CEL expression that determines whether all the resources are in their desired state before running checks on them. 
+        It receives a \`resources\` array of the static and non-static resources.
+        The default behavior is to wait until all the resources are healthy \`dyn(resources).all(r, k8s.isHealthy(r))\`.`,
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:94]: I think we should be wating for ready, not for healthy

File: canary-checker/docs/reference/1-kubernetes-resource.mdx (hunk 1 above)
Comment (adityathebe) [reply canary-checker/docs/reference/1-kubernetes-resource.mdx:130]: @moshloop it's not supported in non-static resources as well.

File: canary-checker/docs/reference/1-kubernetes-resource.mdx (hunk 2 above)
Comment (adityathebe) [reply canary-checker/docs/reference/1-kubernetes-resource.mdx]: A fixture that showcases templating is in this PR https://github.com/flanksource/canary-checker/pull/1810

File: canary-checker/docs/reference/1-kubernetes-resource.mdx (hunk 1 above)
Comment (moshloop) [reply canary-checker/docs/reference/1-kubernetes-resource.mdx:131]: ok

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+The `resources` and  `staticResources` fields can be templated using <CommonLink to="gotemplate">Go Templates</CommonLink>.
+This is helpful in creating resources with random names. 
+Example: you can setup a resource to create a pod with random name on each check run. This way you don't have to wait for the pod to be deleted on every check.
+
+:::info
+Templating the **Group**, **Version**, **Kind** & **Namespace** however isn't allowed.
+:::
+
+<details summary="Templating resources">
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:134]:
```suggestion
<details summary="Creating a pod with a unique name on every run">
```

File: canary-checker/docs/reference/1-kubernetes-resource.mdx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+# <Icon name="k8s" /> Kubernetes Resource
+
+The Kubernetes resource check creates kubernetes resources based on the provided manifests & perform checks on them. Some common use case of this
+check would be to see if a service is accessible via the ingress as shown in the example below.
+
+```yaml title="ingress_test.yaml"  file=../../../modules/canary-checker/fixtures/k8s/kubernetes_resource_ingress_pass.yaml
+
+```
+
+:::warning
```
Comment (moshloop) [inline canary-checker/docs/reference/1-kubernetes-resource.mdx:15]: Can we create a minimal version of the of the check without a namespace specified, and then move this example + warning to an example called "Creating a new namespace for a check"

**Pull request merged**